[pytorch] factor out return_names api

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47438 [pytorch] split out trace type generator and migrate to new codegen model
* **#47437 [pytorch] factor out return_names api**
* #47436 [pytorch] TraceType codegen tweak - newline before redispatch call

Differential Revision: [D24808213](https://our.internmc.facebook.com/intern/diff/D24808213)

ljk53 committed:
[pytorch] factor out return_names api

[ghstack-poisoned]

ezyang reviewed (approved):
Assuming no substantive code changes in move

ljk53 committed:
Update on "[pytorch] factor out return_names api"




[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out return_names api"


Differential Revision: [D24808213](https://our.internmc.facebook.com/intern/diff/D24808213)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] factor out return_names api"


Differential Revision: [D24808213](https://our.internmc.facebook.com/intern/diff/D24808213)

[ghstack-poisoned]
